clash-lib/src/Clash/Core/Type.hs

```diff
@@ -536,6 +537,11 @@ reduceTypeFamily tcm (tyView -> TyConApp tc tys)
 
 reduceTypeFamily _ _ = Nothing
 
+isUnresolvedTypeFamily ::  TyConMap -> Type -> Bool
+isUnresolvedTypeFamily tcm (tyView -> TyConApp tcNm _args)
```

Comment:
Shouldn't this be `coreView`? Since `coreView` actually tries to solve type families. So If after `coreView` we still see a type family, that means it's stuck.

Reply:
Aha, that makes sense. Right now I was banking on the transformations reducing the type family at some point (after which `caseElemNonReachable` could do its job).

Reply:
Fixed, thank you.

Reply:
Looking better at how `stuckTypeFamily` is actually used, I see that you're passing in types that have already gone through `coreView`, so it's kinda wasteful to call `coreView` again. I guess you should just call it `isTypeFamilyApplication` (using just `tyView`), and then inside `solveEq` add a comment that any remaining type family applications are stuck ones, as they should've been solved by `coreView` if they werent stuck.

Reply:
Yeah.. I thought I'd do it this way to make my intent very clear, but I comment will suffice too. Thanks!